cli/admin/src/main/java/io/pravega/cli/admin/readerGroup/ParseReaderGroupStreamCommand.java

```diff
@@ -0,0 +1,173 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.cli.admin.readerGroup;
+
+import io.pravega.cli.admin.AdminCommand;
+import io.pravega.cli.admin.AdminCommandState;
+import io.pravega.cli.admin.CommandArgs;
+import io.pravega.cli.admin.segmentstore.ReadSegmentRangeCommand;
+import io.pravega.cli.admin.utils.ConfigUtils;
+import io.pravega.cli.admin.utils.FileHelper;
+import io.pravega.client.admin.impl.ReaderGroupManagerImpl;
+import io.pravega.client.control.impl.Controller;
+import io.pravega.client.state.impl.UpdateOrInitSerializer;
+import io.pravega.client.stream.impl.StreamSegments;
+import io.pravega.controller.server.SegmentHelper;
+import io.pravega.controller.server.security.auth.GrpcAuthHelper;
+import io.pravega.shared.NameUtils;
+import io.pravega.shared.protocol.netty.PravegaNodeUri;
+import io.pravega.shared.protocol.netty.WireCommandType;
+import io.pravega.shared.protocol.netty.WireCommands;
+import lombok.Cleanup;
+import lombok.val;
+import org.apache.curator.framework.CuratorFramework;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileInputStream;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+
+public class ParseReaderGroupStreamCommand extends AdminCommand {
+
+    private final static int HEADER = WireCommands.TYPE_SIZE;
+    private final static int LENGTH = WireCommands.TYPE_PLUS_LENGTH_SIZE - WireCommands.TYPE_SIZE;
+    private final static int TYPE = WireCommandType.EVENT.getCode();
+    private final static int REQUEST_TIMEOUT_SECONDS = 10;
+
+    private final GrpcAuthHelper authHelper;
+
+    public ParseReaderGroupStreamCommand(CommandArgs args) {
+        super(args);
+
+        authHelper = new GrpcAuthHelper(true, "secret", 600);
+    }
+
+    @Override
+    public void execute() throws Exception {
+        ensureArgCount(4);
+        final String scope = getArg(0);
+        final String readerGroup = getArg(1);
+        final String segmentStoreHost = getArg(2);
+        final String fileName = getArg(3);
+        String stream = NameUtils.getStreamForReaderGroup(readerGroup);
+
+        @Cleanup
+        CuratorFramework zkClient = createZKClient();
+        @Cleanup
+        SegmentHelper segmentHelper = instantiateSegmentHelper(zkClient);
+        String tmpfilename = "tmp/RG_" + readerGroup + "_" + new Random().nextInt(1000);
```

Comment:
Maybe a timestamp is better instead of a random because i) you ensure that you are not going to overwrite the file (now you have 0.001% of odds that this happens), and ii) you know easily by name the chronological order in which multiple files have been created.

Reply:
done